### PR TITLE
Resolve missing Input Leap XDG desktop icon

### LINKED ISSUE #1802: Incorrect app id matching for .desktop icon in Wayland
### What happened?

In Wayland session the app shows generic icon. To correctly display the icon in Wayland session desktop file should provide correct app id. IUUC it should be named like com.github.input-leap.desktop or contain "StartupWMClass=com.github.input-leap" in "Desktop entry". Not sure what's the correct way to make it proper.

Before:

Screenshot from 2023-12-21 01-34-23

After:

Screenshot from 2023-12-21 01-39-35

### Version

From Git HEAD or commit (specify below)

### Git commit hash (if applicable)

3e681454

### If applicable, where did you install Input Leap from?

self built install with patches to allow to run in native Wayland.

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### If applicable, are you using Wayland or X11?

Wayland

### What OS versions are you using?

Arch Linux, GNOME 45

### Relevant log output

_No response_

### Any other information

_No response_

### PR DESCRIPTION
**Pull request by shymega**
Also, simplify CMake conditionals, legacy code.

Should fix #1802.

## Contributor Checklist:

* [ ] I have created a file in the `doc/newsfragments` directory *IF* it is a
      user-visible change (and make sure to read the `README.md` in that directory)

### CODE REVIEW

File: res/input-leap.desktop [hunk 1]
```diff
@@ -3,7 +3,8 @@ Type=Application
 Name=Input Leap
 Comment=Keyboard and mouse sharing solution
 Exec=input-leap
-Icon=input-leap
+Icon=input-leap.svg
```
Comment (sithlord48) [inline res/input-leap.desktop:6]: The icon name without svg should work correctly when inputleap is installed.

**Comment by darkblaze69** [comment]
This still doesn't work - just generic icon.

Screenshot from 2023-12-21 01-34-23

In GNOME looking glass it appears as com.github.input-leap. If I specify `StartupWMClass=com.github.input-leap` in desktop file or rename it to `com.github.input-leap.desktop`, it appears correctly.

Screenshot from 2023-12-23 16-40-56

**After renaming to com.github.input-leap.desktop, it appears correctly in looking glass:**

Screenshot from 2023-12-23 16-46-31

**Comment by sithlord48** [comment]
Can you try using the name `io.github.input_leap.InputLeap` this is what our appstream name would be. Have you installed inputleap ?

**Comment by darkblaze69** [comment]
Yes, I installed it with this patch, StartupWMClass is as in patch, but it's still generic icon.

```
grep -ri leap /usr/share/applications/
/usr/share/applications/input-leap.desktop:Name=Input Leap
/usr/share/applications/input-leap.desktop:Exec=input-leap
/usr/share/applications/input-leap.desktop:Icon=input-leap.svg
/usr/share/applications/input-leap.desktop:StartupWMClass=io.github.input_leap.InputLeap
```

Also doesn't work if I rename this file to `io.github.input_leap.InputLeap.desktop`

Does it show for you the correct icon if run in GNOME wayland by
`DISPLAY= input-leap`

**Comment by sithlord48** [comment]
the icon name should match the desktop name also the icon should be io... also

**Comment by darkblaze69** [comment]
does this patch work for you practically?

**Comment by sithlord48** [comment]
I currently do have not have a wayland session with a DE that has a working portal so i can't really tell you only that in other apps i've done with flatpaks they use the appsteam name and these seam to be also what wayland is looking for along with an icon somewhere in the current theme icon theme or in the generic hicolor theme.

File: res/input-leap.desktop (hunk 1 above)
Comment (shymega) [reply res/input-leap.desktop:6]: It should, but I wanted to see if adding the extension made a difference.

**Comment by shymega** [comment]
I searched the codebase for 'com.github.*`, and not much appeared except for Flatpak and Snap.

We should be using `io.github.input_leap.InputLeap`. Bizarre.

**Comment by shymega** [comment]
Oh. I see now. It's Qt doing it. This is a simple fix. I'll do it now.

File: res/input-leap.desktop (hunk 1 above)
Comment (shymega) [reply res/input-leap.desktop]: Reverted.

**Comment by shymega** [comment]
@darkblaze69 Hey - would you be able to test the latest (force-pushed) commit? I think I've found the issue.

**Comment by darkblaze69** [comment]
Still not working. But now looking glass shows `io.github.input-leap`

image

**Comment by shymega** [comment]
Apologies, accidentally deleted the branch. Restored. I've asked @sithlord48 on IRC about this, as they're more experienced with Qt than I am. I think I'm on the right track.

**Comment by sithlord48** [comment]
Tldr this patch needs todo the following
1. rename the icon to io.github.input_leap.InputLeap.svg (when installing on linux) 
2. Use this same name for the desktop file (
3. In `/src/gui/src/MainWindow.cpp` In the constructor around line 140 add 
```
#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
    qApp()->setDesktopFileName(QStringLiteral("io.github.input_leap.InputLeap");
#endif
```

This will only work with Qt 5.15 + . And the Desktop file name should match our Rev DNS name from the appstream spec (we should also consider shipping a manifest to be fully appsteam compliant but that is a patch for another day )

**Comment by sithlord48** [comment]
For some reason i was not able to add to this branch so i had to make one on my fork 
https://github.com/input-leap/input-leap/pull/1809 should replace this and be correct @darkblaze69  can you please test this is working correctly for you now.

**Comment by darkblaze69** [comment]
thanks, the new PR looks good in Wayland.
